src/System.Net.Http/src/System/Net/Http/MessageProcessingHandler.cs

```diff
@@ -12,6 +13,22 @@ namespace System.Net.Http
 {
     public abstract class MessageProcessingHandler : DelegatingHandler
     {
+        // Private class used to capture the SendAsync state in
+        // a closure, while simultaneously avoiding a tuple allocation.
+        private class SendState : TaskCompletionSource<HttpResponseMessage>
+        {
+            public MessageProcessingHandler self;
```

Comment:
Would something like `Handler` be a better name for this (if so, update the ctor parameter to match)?